components/timeout-transition-group.jsx

```diff
@@ -0,0 +1,241 @@
+/**
+ * The CSSTransitionGroup component uses the 'transitionend' event, which
+ * browsers will not send for any number of reasons, including the
+ * transitioning node not being painted or in an unfocused tab.
+ *
+ * This TimeoutTransitionGroup instead uses a user-defined timeout to determine
+ * when it is a good time to remove the component. Currently there is only one
+ * timeout specified, but in the future it would be nice to be able to specify
+ * separate timeouts for enter and leave, in case the timeouts for those
+ * animations differ. Even nicer would be some sort of inspection of the CSS to
+ * automatically determine the duration of the animation or transition.
+ *
+ * This is adapted from Facebook's CSSTransitionGroup which is in the React
+ * addons and under the Apache 2.0 License.
```

Comment:
k, I added the note to the top of the file: https://github.com/mmmavis/teach.webmaker.org/commit/1093ac9b2d2118048997e726f3739cf5c8f2c71c#diff-bee7c6d20d31386bfb4d4c7c6667f490L1